lib/internal/test_runner/test.js

```diff
@@ -414,6 +418,9 @@ class Test extends AsyncResource {
     validateOneOf(name, 'hook name', kHookNames);
     // eslint-disable-next-line no-use-before-define
     const hook = new TestHook(fn, options);
+    if (name === 'before') {
+      hook.run = runOnce(hook.run);
+    }
```

Comment:
```suggestion
    if (name === 'before' || name === 'after') {
      fn = runOnce(fn);
    }
    const hook = new TestHook(fn, options);
```

Reply:
I think we can also wrap `after` hooks and replace 
```js
 const after = runOnce(async () => {
      if (this.hooks.after.length > 0) {
        await this.runHook('after', { args, ctx });
      }
  });
```

Reply:
I avoided wrapping the function directly at this point because it might not be a function yet, as the constructor handles this case:
https://github.com/nodejs/node/blob/4e09df4dc8851a0cdd935e05052db4e2668a8b54/lib/internal/test_runner/test.js#L174-L176

Do you think the same handling should be added here to wrap `fn` directly?

> I think we can also wrap after hooks and replace

I believe this would change the behavior of the `after` hook though:
Currently, if you have multiple `after` hooks, and one of them throws, we will not re-run any of them in the `catch` block, where as by changing to wrapping each hook individually we will run all the hooks that have not yet been executed.